cudf/cudaarray.py

```diff
@@ -0,0 +1,300 @@
+"""Implements CudaNDArray with the same API as
```

Comment:
I am concerned about completely re-implementing `numba.cuda.DeviceNDArray` just to create an array that uses a different backend for memory. I requested that Numba add the ability to plug in a different backend for memory in `numba.cuda.DeviceNDArray` when I had to similarly override Numba APIs so that cudf could use RMM. That RFE is here: https://github.com/numba/numba/issues/3247

I already feel uncomfortable about RMM having to re-implement those APIs, I would hate to have it in multiple places. Perhaps instead we could put resources behind adding the functionality to Numba?